libs/payments/capability/src/lib/capability.manager.ts

```diff
@@ -2,9 +2,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { ContentfulManager } from '../../../../shared/contentful/src';
 import { Injectable } from '@nestjs/common';
 
+import { ClientIdCapabilityResult } from './capability.types';
+
 @Injectable()
 export class CapabilityManager {
-  constructor() {}
+  constructor(private contentfulManager: ContentfulManager) {}
+
+  /**
+   * Fetch the list of capabilities for the given price ids.
+   */
+  async planIdsToClientCapabilities(
+    subscribedPrices: string[]
+  ): Promise<ClientIdCapabilityResult> {
+    if (!subscribedPrices.length) return [];
+
+    const detailsResult =
+      await this.contentfulManager.getPurchaseDetailsForCapabilityService([
+        ...subscribedPrices,
+      ]);
+
+    const result: ClientIdCapabilityResult[] = [];
+
+    for (const subscribedPrice of subscribedPrices) {
+      const fromOffering = detailsResult.offeringForPlanId(subscribedPrice);
+      if (!fromOffering) continue;
+
+      result.push({ subscribedPrice });
+    }
+
+    return result;
```

Comment:
This seems a little strange, since the return value of this function would look like:

```
[{
  subscribedPrice: "asdf"
}, {
  subscribedPrice: "asdf2"
}]
```

It would seem like this function should return capabilities given the function's name?

Reply:
Fixed. Thanks for the pair!